aten/src/ATen/native/cuda/MaxUnpooling.cu

```diff
@@ -98,35 +98,219 @@ __global__ void max_unpooling3d_backward_kernel(
   }
 }
 
+static void max_unpooling2d_shape_check(
```

Comment:
Maybe move this to Pool.h. There's already some utility shape checking code there for other pooling functions. Then call it on both CPU and CUDA